documentation/experimental-features.md

```diff
@@ -0,0 +1,83 @@
+# Experimental features: description and process
```

Comment:
Nothing new, just a move of file from https://github.com/meilisearch/engine-team to here